Sprint 14

Anna-Peres opened the pull request.

Anna-Peres committed:
Тема 2

Anna-Peres committed:
Тема 2

Anna-Peres committed:
Тема 2

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
Экран создания трекера «Привычка» реализован

Anna-Peres committed:
Создан экран "Новая привычка"

Anna-Peres committed:
no message

Anna-Peres committed:
Cоздан экран "Расписание"

Anna-Peres committed:
Тема 1

Anna-Peres committed:
Добавлена CollectionView

Anna-Peres committed:
Добавлено поле для поиска

Anna-Peres committed:
no message

Anna-Peres committed:
Добавление выделенных дней недели в массив

Anna-Peres committed:
Скорректировано поле для поиска

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
Доделан функционал экрана с расписанием

Anna-Peres committed:
no message

Anna-Peres committed:
Реализовано добавление нового трекера

Anna-Peres committed:
no message

Anna-Peres committed:
no message

Anna-Peres committed:
Задание выполнено

Kirchberg reviewed (changes requested):
⚠️ Критические замечания (это нужно исправить обязательно):

1. В NewHabitViewController после ввода текста в UITextField клавиатуру можно скрыть только нажатием return. Отсутствует возможность скрыть ее по тапу на свободную область экрана. Аналогичная проблема с UISearchBar на главном экране. Это делает использование приложения неудобным. Пожалуйста, добавь обработку касания вне текстовых полей для скрытия клавиатуры.
2. На главном экране отстутствует разделитель у таб бара и не совпадает «жирность» шрифта в слове «Трекеры» (рекомендую перепроверить, точно ли верная степень используется), [см. скриншот сравнения симулятора с макетом](https://ibb.co/XrrpRPnc).

Без исправлений этих пунктов, я не смогу принять у тебя работу. Если возникнут вопросы, обязательно задавай их Наставнику!

--- 

🍏 Рекомендации (это можно исправить по желанию):

1. В TrackersViewController в методе collectionView(:viewForSupplementaryElementOfKind:at:) заголовок секции жестко закодирован: view.titleLabel.text = "Новая категория". Из-за этого все категории трекеров отображаются с одинаковым названием. Необходимо использовать реальное название категории из модели данных: filteredCategories[indexPath.section].title. Не забудь в следующем спринте это поправить ✨
2. В NewHabitViewController и ScheduleViewController используется устаревший подход для создания ячеек таблицы с проверкой if let reusedCell. Рекомендую зарегистрировать класс ячейки в viewDidLoad (tableView.register(NewHabitCell.self, forCellReuseIdentifier: "...")) и затем использовать tableView.dequeueReusableCell(withIdentifier: "...", for: indexPath), что сделает код более современным и кратким.
3. Метод viewDidLoad в TrackersViewController довольно большой. Можно сгруппировать вызовы методов настройки UI (addLargeTitleLabel, addPlusButton и т.д.) в один общий метод, например setupUI(), для улучшения читаемости.

--- 

🍪 Для размышлений 

1. Сейчас для хранения данных используются массивы в TrackersViewController. В следующем спринте мы будем переходить на Core Data. Можно уже сейчас подумать, как изменится архитектура: где будет происходить загрузка данных, как UI будет реагировать на изменения в базе.
2. Для фильтрации трекеров по дате и тексту поиска сейчас пересоздается массив filteredCategories. Для больших объемов данных это может быть неэффективно. В будущем стоит изучить NSFetchedResultsController, который оптимизирован для работы с Core Data и автоматического обновления UI.

---

Чтобы твоя работа была зачтена, необходимо исправить все критические замечания.
Жду исправлений :)

Anna-Peres committed:
- Добавлена обработка касания вне текстовых полей.
- Исправлена толщина шрифта слова "Трекеры".
- Добавлен разделитель у таббара.
- Название категории теперь используется из модели.
- Исправлен подход для создания ячеек таблицы.
- Методы добавления элементов UI в контроллеры объединены в setupUI().

Kirchberg reviewed (approved):
Не забудь вмерджить пул-реквест в основную ветку.
Успехов в следующем задании!
